pkg/workflows/management/create_cluster.go

```diff
@@ -0,0 +1,379 @@
+	if err != nil {
+		commandContext.SetError(err)
+		return &workflows.CollectDiagnosticsTask{}
+	}
+	commandContext.WorkloadCluster = workloadCluster
+
+	if !commandContext.BootstrapCluster.ExistingManagement {
+		logger.Info("Creating EKS-A namespace")
+		err := commandContext.ClusterManager.CreateEKSANamespace(ctx, commandContext.WorkloadCluster)
+		if err != nil {
+			commandContext.SetError(err)
+			return &workflows.CollectDiagnosticsTask{}
+		}
+
+		logger.Info("Installing cluster-api providers on workload cluster")
+		err = commandContext.ClusterManager.InstallCAPI(ctx, commandContext.ClusterSpec, commandContext.WorkloadCluster, commandContext.Provider)
+		if err != nil {
+			commandContext.SetError(err)
+			return &workflows.CollectDiagnosticsTask{}
+		}
+
+		logger.Info("Installing EKS-A secrets on workload cluster")
+		err = commandContext.Provider.UpdateSecrets(ctx, commandContext.WorkloadCluster, commandContext.ClusterSpec)
+		if err != nil {
+			commandContext.SetError(err)
+			return &workflows.CollectDiagnosticsTask{}
+		}
+	}
+
+	return &installResourcesOnManagementTask{}
+}
+
+func (s *createWorkloadClusterTask) Name() string {
+	return "workload-cluster-init"
+}
+
+func (s *createWorkloadClusterTask) Restore(ctx context.Context, commandContext *task.CommandContext, completedTask *task.CompletedTask) (task.Task, error) {
+	return nil, nil
+}
+
+func (s *createWorkloadClusterTask) Checkpoint() *task.CompletedTask {
+	return nil
+}
+
+func (s *installResourcesOnManagementTask) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	if commandContext.BootstrapCluster.ExistingManagement {
+		return &moveClusterManagementTask{}
+	}
+	logger.Info("Installing resources on management cluster")
+
+	if err := commandContext.Provider.PostWorkloadInit(ctx, commandContext.WorkloadCluster, commandContext.ClusterSpec); err != nil {
+		commandContext.SetError(err)
+		return &workflows.CollectDiagnosticsTask{}
+	}
+	return &moveClusterManagementTask{}
+}
+
+func (s *installResourcesOnManagementTask) Name() string {
+	return "install-resources-on-management-cluster"
+}
+
+func (s *installResourcesOnManagementTask) Restore(ctx context.Context, commandContext *task.CommandContext, completedTask *task.CompletedTask) (task.Task, error) {
+	return nil, nil
+}
+
+func (s *installResourcesOnManagementTask) Checkpoint() *task.CompletedTask {
+	return nil
+}
+
+func (s *moveClusterManagementTask) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	if commandContext.BootstrapCluster.ExistingManagement {
+		return &installEksaComponentsOnWorkloadTask{}
+	}
+	logger.Info("Moving cluster management from bootstrap to workload cluster")
+	err := commandContext.ClusterManager.MoveCAPI(ctx, commandContext.BootstrapCluster, commandContext.WorkloadCluster, commandContext.WorkloadCluster.Name, commandContext.ClusterSpec, types.WithNodeRef())
```

Comment:
do we need to pause the eks-a reconciliation before we do this?
if not, you take the risk of the eks-a controller trying to create a new capi cluster

basically do the same thing that clusterctl does to move resources

Reply:
We can add the pause reconcile annotation here. However, since the order of tasks is that we move CAPI first, and then install EKSA controller, the capi cluster will already exist by the time the controller starts reconciliation, so would it make a difference?